LUN-362: Add threshold for finished exercises

SteffiStuffel opened the pull request:
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

f1sh1918 reviewed (commented):
Looks fine for me.
Just recognized one issue probably not connected to that.
1.Enable dev mode -> fail all -> correct finish screen
2. retry solve all -> again failing screen :/
![image](https://user-images.githubusercontent.com/37902063/187685727-4331686d-ffa7-41de-8f5f-56c4a637280d.png)

steffenkleinle commented on src/services/helpers.ts:
```diff
@@ -166,3 +165,6 @@ export const calculateScore = (documentsWithResults: DocumentResult[]): number =
 }
+
+export const willNextExerciseUnlock = (previousScore: number | undefined, score: number): boolean =>
+  score > SCORE_THRESHOLD_UNLOCK && (previousScore === undefined || previousScore <= SCORE_THRESHOLD_UNLOCK)
```
Optional:
```suggestion
  score > SCORE_THRESHOLD_UNLOCK && previousScore ?? 0 <= SCORE_THRESHOLD_UNLOCK)
```

steffenkleinle reviewed (commented):
Tested on android.

I am getting a warning about a require cycle helpers <-> async storage. Not sure if this was introduced in this PR but definitely related to calculation of the score and should be fixed.

steffenkleinle commented:
Please add a release note :)

SteffiStuffel commented:
> Looks fine for me. Just recognized one issue probably not connected to that. 1.Enable dev mode -> fail all -> correct finish screen 2. retry solve all -> again failing screen :/
> 

Actually this was fine, as cheat mode (also in when solving) "needs" three tries, which is to bad to unlock. I adjusted the cheat mode, so it only needs one try when solve all

Removed require cycle, good catch, thanks.

added release note

f1sh1918 commented:
@SteffiStuffel pls resolve merge conflicts and i will rereview
